diff --git a/arch/arm/mach-msm/board-8960-display.c b/arch/arm/mach-msm/board-8960-display.c
index 047684b..65ec7a6 100644
--- a/arch/arm/mach-msm/board-8960-display.c
+++ b/arch/arm/mach-msm/board-8960-display.c
@@ -588,7 +588,7 @@ static struct msm_panel_common_pdata mdp_pdata = {
 	.mem_hid = MEMTYPE_EBI1,
 #endif
-	.cont_splash_enabled = 0x01,
+	.cont_splash_enabled = 0x00,
 	.splash_screen_addr = 0x00,
 	.splash_screen_size = 0x00,
 	.mdp_iommu_split_domain = 0,
