diff --git a/hw/xfree86/fbdevhw/fbdevhw.c b/hw/xfree86/fbdevhw/fbdevhw.c
--- a/hw/xfree86/fbdevhw/fbdevhw.c
+++ b/hw/xfree86/fbdevhw/fbdevhw.c
@@ -335,8 +335,12 @@
         char *sysfs_path = NULL;
         char *node = strrchr(dev, '/') + 1;
 
-        if (asprintf(&sysfs_path, "/sys/class/graphics/%s/device/subsystem", node) < 0 ||
-            readlink(sysfs_path, buf, sizeof(buf) - 1) < 0 ||
+        if (asprintf(&sysfs_path, "/sys/class/graphics/%s/device/subsystem", node) < 0) {
+            close(fd);
+            return -1;
+        }
+
+        if (readlink(sysfs_path, buf, sizeof(buf) - 1) >= 0 &&
             strstr(buf, "bus/pci")) {
             free(sysfs_path);
             close(fd);
